packages/kit/types/ambient-modules.d.ts

```diff
@@ -15,6 +15,10 @@ declare module '$app/env' {
 	 * `true` when prerendering, `false` otherwise.
 	 */
 	export const prerendering: boolean;
+	/**
+	 * The Vite.js mode the app is running in. Configure in `config.kit.vite.mode`. By default, `svelte-kit dev` runs in `development` mode and `svelte-kit build` runs in `production` mode.
```

Comment:
it's not quite clear from these docs and the docs above for `dev` how the two are different from each other

Reply:
It is technically redundant, but only because we'd be reflecting Vite's behaviour and Vite's behaviour is redundant https://vitejs.dev/guide/env-and-mode.html#env-variables. In practice I think it's beneficial to have a boolean `dev` value, and if people have found that it's valuable to expose `mode` then it makes sense to have both i think

Reply:
@benmccann Yes, I should expand on these docs. Mode is used to infer `.env.[mode]` dotenv files to load. As these values are statically replaced during a build, we can utilise `mode` to load the config for our target env.

While this is similar to `dev` I'd argue not redundant. As I raised in https://github.com/sveltejs/kit/pull/1613#discussion_r642819593 there are many instances for building and running a SvelteKit app and in each instance I may want to use different **config** and conditionally execute code when `dev=false` in `prod` builds.

EG:
| command                                                           | purpose                                                                                           | envs                                     |
| ----------------------------------------------------------------- | ------------------------------------------------------------------------------------------------- | ---------------------------------------- |
| `svelte-kit dev`                                                  | run **dev** server against **local** resources                                                    | `NODE_ENV=development`                   |
|                                                                   |                                                                                                   | `VITE_MODE=development` (default)        |
|                                                                   |                                                                                                   | `env.development`                        |
|                                                                   |                                                                                                   | `dev=true`                               |
|                                                                   |                                                                                                   |                                          |
| run **production** against **prod** resources                     | `svelte-kit build`                                                                                | `NODE_ENV=production`                    |
|                                                                   |                                                                                                   | `VITE_MODE=production` (default)         |
|                                                                   |                                                                                                   | `env.production`                         |
|                                                                   |                                                                                                   | `dev=false`                              |
|                                                                   |                                                                                                   |                                          |
| run **production** build locally against **local** resources      | `svelte-kit preview`                                                                              | `NODE_ENV=production`                    |
|                                                                   |                                                                                                   | `VITE_MODE=production` (default) :x:     |
|                                                                   | We want to load `env.development` here but default `VITE_MODE=production` was used during `build` | `env.production` :x:                     |
|                                                                   | not useful to resolve issue                                                                       | `dev=false`                              |
|                                                                   |                                                                                                   |                                          |
| build & deploy to **staging** against **staging** resources       | `svelte-kit build`                                                                                | `NODE_ENV=production`                    |
|                                                                   |                                                                                                   | `VITE_MODE=production` (default) :x:     |
|                                                                   | We want to load `env.staging` here but default `VITE_MODE=production` was used during build       | `env.production` :x:                     |
|                                                                   | not useful to resolve issue                                                                       | `dev=false`                              |
|                                                                   |                                                                                                   |                                          |
| build & deploy to **production** against **production** resources | `svelte-kit build`                                                                                | `NODE_ENV=production`                    |
|                                                                   |                                                                                                   | `VITE_MODE=production` (default) :check: |
|                                                                   |                                                                                                   | `env.production` :check:                 |
|                                                                   |                                                                                                   | `dev=false`                              |
|                                                                   |                                                                                                   |                                          |

Reply:
That's a very long answer I don't quite know how to digest :smile:  Basically my question is are these representing different values. E.g. one is `NODE_ENV` and one is `VITE_MODE`? We should document what underlying flag these each represent

Reply:
Yes. One is essentially `NODE_ENV`, one is `VITE_MODE`. As far as I understand `dev` cannot represent all the states that `mode` can, and therefore is not redundant.